### PR TITLE
Alfoa/dataobject rework ensemble model

### PR DESCRIPTION
**Pull request by alfoa**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
This fixes the ensemble models (not time series) and the database serialization problem.

##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

**Comment by alfoa** [comment]
On FALCON:
```
Ran 477 tests in 499.5 seconds
352 passed, 59 skipped, 0 pending, 125 FAILED
```

**Comment by alfoa** [comment]
```
**** THE FOLLOWING TESTS CHANGED OUTCOMES ****
**** Test framework.dists_vars | has changed outcome from <OK> to <FAILED> !!!!
**** Test framework.pcaResponseSurface | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/aliasSystemTests.testEnsembleModelAliased | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/ensembleModelTests.testEnsembleModelFullyCorrVars | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/ensembleModelTests.testEnsembleModelLateralInsertions | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/ensembleModelTests.testEnsembleModelLinearParallel | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/ensembleModelTests.testEnsembleModelLinearThread | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/ensembleModelTests.testEnsembleModelNonLinearParallel | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/ensembleModelTests.testEnsembleModelNonLinearThread | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/ensembleModelTests.testEnsembleModelWith2CodesAndAlias | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/ensembleModelTests.testEnsembleModelWithCode | has changed outcome from <FAILED> to <OK> !!!!
```
Summary

```
*****************
Number of success first file:
343
Number of success second file:
352
NUMBER OF SUCCESSES INCREASED: +9
**** NUMBER OF TESTS IN THE TWO FILES ARE DIFFERENT: ****
**** first file :532 /= second file: 536****
TOTAL TIME OF TESTS DECREASED       : reference_time(536.3) > new_time(499.5)
AVERAGE TIMES FOR SUCCESS TESTS ARE : reference_time(1.56355685131) v.s. new_time(1.41903409091)
```

File: framework/Models/EnsembleModel.py [hunk 1]
```diff
@@ -432,69 +432,36 @@ def collectOutput(self,finishedJob,output):
...
-    exportDictTargetEvaluation = {}
-    outcomes, targetEvaluations, optionalOutputs = out
+    outcomes, targetEvaluations, optionalOutputs = evaluation[1]
     try:
       jobIndex = self.tempOutputs['uncollectedJobIds'].index(finishedJob.identifier)
       self.tempOutputs['uncollectedJobIds'].pop(jobIndex)
     except ValueError:
       jobIndex = None
+
+    #joinedGeneralMetadata = {}
```
Comment (wangcj05) [inline framework/Models/EnsembleModel.py]: remove

File: framework/Models/EnsembleModel.py [hunk 2]
```diff
@@ -432,69 +432,36 @@ def collectOutput(self,finishedJob,output):
...
       if jobIndex is not None:
         for optionalModelOutput in self.modelsDictionary[modelIn]['OutputObject']:
-          optionalModelOutput.addRealization(returnDict[modelIn]['response'])
-          #self.modelsDictionary[modelIn]['Instance'].collectOutput(finishedJob,optionalModelOutput,options={'exportDict':copy.copy(optionalOutputs[modelIn])})
+          optionalModelOutput.addRealization(optionalOutputs[modelIn])
+      # update the responses
+      joinedResponse.update(outcomes[modelIn]['response'])
+
+      # get general metadata
+      #joinedGeneralMetadata.update(outcomes[modelIn]['general_metadata'])
```
Comment (wangcj05) [inline framework/Models/EnsembleModel.py]: remove?

File: framework/Models/EnsembleModel.py (hunk 1 above)
Comment (alfoa) [reply framework/Models/EnsembleModel.py]: This is a placeholder for me now to remember to readd the general metadata

File: framework/Models/EnsembleModel.py (hunk 2 above)
Comment (alfoa) [reply framework/Models/EnsembleModel.py]: as above

**Comment by alfoa** [comment]
@wangcj05 @PaulTalbot-INL any other comments? can we merge?

File: tests/framework/gold/dists_vars/PrintPointSet_dump.csv [hunk 3]
```diff
@@ -1,2 +1,2 @@
 t4,t2,t3,t1,y1,y0,x0,x1,z0,z1,out
-2564.95763481,949.121133823,2564.95763481,2214.56454928,60.0370545611,60.0370545611,0.80235439175,0.80235439175,60.0370545611,60.0370545611,0.0
+1638.00937531,717.896438963,1638.00937531,1816.20154577,71.8212094004,70.5648376018,0.713709826185,0.492427456546,70.5648376018,71.8212094004,0.0
```
Comment (wangcj05) [inline tests/framework/gold/dists_vars/PrintPointSet_dump.csv]: Why all the values are changed?

Comment (wangcj05) [reply tests/framework/gold/dists_vars/PrintPointSet_dump.csv]: I figured it  out. The seed is changed.

**Comment by wangcj05** [comment]
For me, it is fine.

**Pull request merged**